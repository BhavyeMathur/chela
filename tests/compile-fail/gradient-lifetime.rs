use redstone_ml::*;

fn main() {
    let _grad = {
        let mut a = Tensor::new([1.0f32, 2.0, 3.0]);
        a.set_requires_grad(true);
        a.backward();

        a.gradient().unwrap()
    };
}
