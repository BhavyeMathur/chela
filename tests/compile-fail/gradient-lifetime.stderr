error[E0597]: `a` does not live long enough
  --> tests/compile-fail/gradient-lifetime.rs:9:9
   |
4  |     let _grad = {
   |         ----- borrow later stored here
5  |         let mut a = Tensor::new([1.0f32, 2.0, 3.0]);
   |             ----- binding `a` declared here
...
9  |         a.gradient().unwrap()
   |         ^ borrowed value does not live long enough
10 |     };
   |     - `a` dropped here while still borrowed
